contrib/nsqd.go

```diff
@@ -0,0 +1,40 @@
+package contrib
+
+import (
+	"github.com/nsqio/nsq/nsqd"
+)
+
+
+type INSQDAddon interface {
+	Enabled() bool
+	Start()
+}
+
+type NSQDAddons struct {
+	addons []INSQDAddon
+}
+
+// Starts all addons that are active
+func (as *NSQDAddons) Start() {
+
+	for _, addon := range as.addons {
+		addon.Start()
+	}
+}
+
+// Initializes addons that have options set
+func NewEnabledNSQDAddons(contribOpts []string, n *nsqd.NSQD) *NSQDAddons {
+	var activeAddons []INSQDAddon
+
+	n.Logf(nsqd.LOG_INFO, "Addons Initializing")
+
+	// ask each addon if it should be initialize
+	dogStats := NewNSQDDogStatsd(contribOpts, n)
+	if dogStats.Enabled() {
```

Comment:
(this comment copied to a new one because the original was hidden by github)

I think it would be better to only call `NewNSQDDogStatsd()` if there are any contribOpts that start with "dogstatsd=" - that way, you can be sure that any "contrib modules" which you did not pass options to did not run anything, without looking at their code.

So this function would just need a little table of name strings and init functions to loop over.